cmd/kubeadm/app/apis/kubeadm/v1beta3/conversion.go

```diff
@@ -39,19 +39,55 @@ func Convert_v1beta3_InitConfiguration_To_kubeadm_InitConfiguration(in *InitConf
 // Convert_v1beta3_ControlPlaneComponent_To_kubeadm_ControlPlaneComponent is required due to the missing ControlPlaneComponent.ExtraEnvs in v1beta3.
 func Convert_v1beta3_ControlPlaneComponent_To_kubeadm_ControlPlaneComponent(in *ControlPlaneComponent, out *kubeadm.ControlPlaneComponent, s conversion.Scope) error {
 	out.ExtraEnvs = []v1.EnvVar{}
+	out.ExtraArgs = convertToArgs(in.ExtraArgs)
 	return autoConvert_v1beta3_ControlPlaneComponent_To_kubeadm_ControlPlaneComponent(in, out, s)
 }
 
 func Convert_kubeadm_ControlPlaneComponent_To_v1beta3_ControlPlaneComponent(in *kubeadm.ControlPlaneComponent, out *ControlPlaneComponent, s conversion.Scope) error {
+	out.ExtraArgs = convertFromArgs(in.ExtraArgs)
 	return autoConvert_kubeadm_ControlPlaneComponent_To_v1beta3_ControlPlaneComponent(in, out, s)
 }
 
 // Convert_v1beta3_LocalEtcd_To_kubeadm_LocalEtcd is required due to the missing LocalEtcd.ExtraEnvs in v1beta3.
 func Convert_v1beta3_LocalEtcd_To_kubeadm_LocalEtcd(in *LocalEtcd, out *kubeadm.LocalEtcd, s conversion.Scope) error {
 	out.ExtraEnvs = []v1.EnvVar{}
+	out.ExtraArgs = convertToArgs(in.ExtraArgs)
 	return autoConvert_v1beta3_LocalEtcd_To_kubeadm_LocalEtcd(in, out, s)
 }
 
 func Convert_kubeadm_LocalEtcd_To_v1beta3_LocalEtcd(in *kubeadm.LocalEtcd, out *LocalEtcd, s conversion.Scope) error {
+	out.ExtraArgs = convertFromArgs(in.ExtraArgs)
 	return autoConvert_kubeadm_LocalEtcd_To_v1beta3_LocalEtcd(in, out, s)
 }
+
+func Convert_v1beta3_NodeRegistrationOptions_To_kubeadm_NodeRegistrationOptions(in *NodeRegistrationOptions, out *kubeadm.NodeRegistrationOptions, s conversion.Scope) error {
+	out.KubeletExtraArgs = convertToArgs(in.KubeletExtraArgs)
+	return autoConvert_v1beta3_NodeRegistrationOptions_To_kubeadm_NodeRegistrationOptions(in, out, s)
+}
+
+func Convert_kubeadm_NodeRegistrationOptions_To_v1beta3_NodeRegistrationOptions(in *kubeadm.NodeRegistrationOptions, out *NodeRegistrationOptions, s conversion.Scope) error {
+	out.KubeletExtraArgs = convertFromArgs(in.KubeletExtraArgs)
+	return autoConvert_kubeadm_NodeRegistrationOptions_To_v1beta3_NodeRegistrationOptions(in, out, s)
+}
+
+func convertToArgs(in map[string]string) []kubeadm.Arg {
+	if in == nil {
+		return nil
+	}
+	args := []kubeadm.Arg{}
+	for k, v := range in {
+		args = append(args, kubeadm.Arg{Name: k, Value: v})
+	}
+	return args
+}
+
+func convertFromArgs(in []kubeadm.Arg) map[string]string {
+	if in == nil {
+		return nil
+	}
+	args := map[string]string{}
+	for _, arg := range in {
+		args[arg.Name] = arg.Value
+	}
+	return args
+}
```

Comment:
Another contributor once asked if we need to add some tests for the `conversion.go`. 

For here, `convertFromArgs` may de-dup some keys, which is expected.

Reply:
i can add some quick unit tests. yes, converting to map drops duplicates.